### PR TITLE
Add option to install specific kernel and boot it by default

### PR DESCRIPTION
**Pull request by imarkov-storpool**
If this looks OK, I will also try to add `kexec` for faster reboots into the new kernel.

### CODE REVIEW

File: roles/2_setup-infra/tasks/subtasks/dependencies-debian.yml [hunk 1]
```diff
@@ -11,6 +11,28 @@
...
+    pkg:
+      - "linux-modules-extra-{{ sp_kernel_version }}"
+    update_cache: yes
+  when:
+    - ansible_distribution == "Ubuntu"
+    - sp_kernel_version is defined
+    - not ansible_kernel | regex_search("^" + sp_kernel_version)
+
+- name: Include tasks for setting default kernel
+  include_tasks: "kernel.yml"
```
Comment (netcho) [inline roles/2_setup-infra/tasks/subtasks/dependencies-debian.yml:34]: Please use a more elaborate naming scheme, for example ``boot_into_updated_kernel`` or something similar.

File: roles/2_setup-infra/tasks/subtasks/dependencies-redhat.yml [hunk 2]
```diff
@@ -18,6 +18,17 @@
...
+- name: Install specified kernel
+  yum:
+    pkg: "kernel-{{ sp_kernel_version }}"
+    update_cache: yes
+  when:
+    - sp_kernel_version is defined
+    - not ansible_kernel | regex_search("^" + sp_kernel_version)
+
+- name: Include tasks for setting default kernel
+  include_tasks: "kernel.yml"
```
Comment (netcho) [inline roles/2_setup-infra/tasks/subtasks/dependencies-redhat.yml:30]: Please use a more elaborate naming scheme, for example, `boot_into_updated_kernel` or something similar.

File: roles/2_setup-infra/tasks/subtasks/kernel.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    reboot_timeout: 600
+    pre_reboot_delay: 0
+    post_reboot_delay: 30
+    test_command: whoami
+  when:
+    - sp_kernel_version is defined
+    - not ansible_kernel | regex_search("^" + sp_kernel_version)
+
+# Not using ansible_kernel since the value does not update after reboot
+- name: Get current kernel
```
Comment (netcho) [inline roles/2_setup-infra/tasks/subtasks/kernel.yml:20]: It's better to make Ansible regather facts, the kernel command line might've changed.

**Review by netcho — CHANGES_REQUESTED** [review]
It looks pretty much OK, I've left some minor comments.

File: roles/2_setup-infra/files/set_default_kernel [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    Usage: $(basename -- "$0") <kernel_version or 0>
+EOUSAGE
+    exit 1
+fi
+kernel_version="$1"
+now="$(date +"%Y-%m-%d_%H-%M-%S")"
+
+DISTRO_REGEX="CentOS|Red Hat Enterprise Linux Server|AlmaLinux|Debian|Ubuntu"
+
+function set_grub_default {
```
Comment (tanev) [inline roles/2_setup-infra/files/set_default_kernel:14]: btw, I know it says #!/bin/bash on the first line, but it is a good idea to always use foo(){} anyway:
http://mywiki.wooledge.org/BashPitfalls#function_foo.28.29

File: roles/2_setup-infra/files/set_default_kernel [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    local default="${1}"
+    local grub_cfg="${2}"
+    if [[ -z "${grub_cfg}" ]]; then
```
Comment (tanev) [inline roles/2_setup-infra/files/set_default_kernel:17]: No need for the quotes inside the [[]]

File: roles/2_setup-infra/files/set_default_kernel [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+for cfg in "/boot/grub/grub.cfg" "/boot/grub2/grub.cfg"; do
+    new_default=$(parse_grub_cfg "$cfg")
+    if [[ -n "$new_default" ]]; then
+        set_grub_default "$new_default" "$cfg"
+        exit 0
+    fi
+done
+
+new_default=$(parse_boot_loader_entries)
```
Comment (tanev) [inline roles/2_setup-infra/files/set_default_kernel:136]: Add quotes here, please.

**Review by netcho — APPROVED** [review]

**Pull request merged**